cache/cache.go

```diff
@@ -0,0 +1,43 @@
+package cache // import "bosun.org/cache"
+
+import (
+	"bosun.org/_third_party/github.com/golang/groupcache/lru"
+	"bosun.org/_third_party/github.com/golang/groupcache/singleflight"
+	"sync"
+)
+
+type Cache struct {
+	g singleflight.Group
+
+	sync.Mutex
+	lru *lru.Cache
+}
+
+type Result struct {
+	Val interface{}
+	Err error
+}
+
+func New() *Cache {
+	return &Cache{
+		lru: lru.New(100),
```

Comment:
agreed.  i didn't do this yet because how to specify the config for this is still an open question.